packages/web3-core/test/unit/web3_request_manager.test.ts

```diff
@@ -379,6 +380,103 @@ describe('Web3RequestManager', () => {
 			});
 		});
 
+		describe('eip15593-provider', () => {
+			beforeEach(() => {
+				jest.spyOn(utils, 'isWeb3Provider').mockReturnValue(false);
+				jest.spyOn(utils, 'isLegacyRequestProvider').mockReturnValue(false);
+				jest.spyOn(utils, 'isLegacySendProvider').mockReturnValue(false);
+				jest.spyOn(utils, 'isLegacySendAsyncProvider').mockReturnValue(false);
+				jest.spyOn(utils, 'isEIP1193Provider').mockReturnValue(true);
+			});
+
+			it('should pass request to provider and resolve if provider resolves it', async () => {
```

Comment:
We should also have some [system tests](https://github.com/ChainSafe/web3.js/issues/5309), and we can plan that for release after alpha.